### PR TITLE
Fix behaviour of build/pull after recent Breeze changes

### PR DESCRIPTION
**Pull request by potiuk**
The behaviour of Breeze after some recent changes related to
pulling and building images in parallel have been slightly broken.
Nothing serious but slightly annoying behaviour:

* when starting breeze shell, the image was attempted to be
  build even if it was not needed (but cache efficiency made it fast
  enough to not be too annoying (unless we updated to newer
  python base image

* breeze pull command for "latest" branch makes no sense any more -
  we stopped pushing "latest" image to ghcr.io, we only push
  cache and "tagged" images. We are now turning --image-tag as
  required in "pull_image" and when someone specifies latest,
  error and helpful message is printed

* --force-build flag in "shell-related-commands" was not
  properly propagated to build-image so it did not actually
  force image building.

All those problems are fixed now.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/breeze/src/airflow_breeze/commands/ci_image_commands.py [hunk 1]
```diff
@@ -605,4 +613,5 @@ def rebuild_or_pull_ci_image_if_needed(
             'Forcing build.[/]'
         )
         ci_image_params.force_build = True
-    run_build_ci_image(verbose, dry_run=dry_run, ci_image_params=ci_image_params, parallel=False)
+    if start_building(ci_image_params=ci_image_params, dry_run=dry_run, verbose=verbose):
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/commands/ci_image_commands.py]: Here we missed a check if image should be built.

File: dev/breeze/src/airflow_breeze/commands/ci_image_commands.py [hunk 2]
```diff
@@ -582,8 +589,9 @@ def rebuild_or_pull_ci_image_if_needed(
         upgrade_to_newer_dependencies=False,
         image_tag=command_params.image_tag,
         platform=command_params.platform,
+        force_build=command_params.force_build,
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/commands/ci_image_commands.py:594]: Here we did not propagate --force-build properly.

**Comment by potiuk** [comment]
Nees #24659  to succeed.

**Comment by potiuk** [comment]
Would be nice to merge this one to get fixed non-rebuild experience

**Review by dstandish — APPROVED** [review]

**Pull request merged**